src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -214,6 +219,9 @@ def first_python_binary() -> Optional[PythonExecutable]:
         path=pex_runtime_env.path,
         interpreter_search_paths=tuple(python_setup.interpreter_search_paths),
         subprocess_environment_dict=subprocess_env_vars.vars,
+        # TODO: This path normalization is duplicated with `engine_initializer.py`. How can we do
+        #  the normalization only once, via the options system?
+        named_caches_dir=Path(global_options.options.named_caches_dir).resolve().as_posix(),
```

Comment:
This is meant to handle using relative paths, which must be expanded to absolute.